src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,26 @@
+package task;
+
+public class Deadline extends Task {
+    private static final String DELIMITER = "/ by";
+
+    public Deadline(String description) {
+        super(description);
+    }
+
+    public String getTypeOfTask() {
+        return "D";
+    }
+
+    public String getFormattedDescription() {
+        String[] slicedDescription = description.split(DELIMITER);
+        try {
+            String what = slicedDescription[0];
```

Comment:
instead of using "what" and "when", maybe use description and time? Should try to name variables as nouns as much as possible.